lib/std/time.zig

```diff
@@ -59,6 +59,398 @@ pub fn sleep(nanoseconds: u64) void {
+
+    pub const Weekday = enum(u3) {
+        monday = 1,
+        tuesday = 2,
+        wednesday = 3,
+        thursday = 4,
+        friday = 5,
+        saturday = 6,
+        sunday = 7,
+
+        pub const names = [_][]const u8{
+            "Monday",
+            "Tuesday",
+            "Wednesday",
+            "Thursday",
+            "Friday",
+            "Saturday",
+            "Sunday",
+        };
+
+        pub fn shortName(w: Weekday) []const u8 {
+            return names[@intFromEnum(w) - 1][0..3];
+        }
+
+        pub fn longName(w: Weekday) []const u8 {
+            return names[@intFromEnum(w) - 1];
+        }
+    };
+
+    pub const Month = enum(u4) {
+        january = 1,
+        february = 2,
+        march = 3,
+        april = 4,
+        may = 5,
+        june = 6,
+        july = 7,
+        august = 8,
+        september = 9,
+        october = 10,
+        november = 11,
+        december = 12,
+
+        pub const names = [_][]const u8{
+            "January",
+            "February",
+            "March",
+            "April",
+            "May",
+            "June",
+            "July",
+            "August",
+            "September",
+            "October",
+            "November",
+            "December",
+        };
+
+        pub fn shortName(m: Month) []const u8 {
+            return names[@intFromEnum(m) - 1][0..3];
+        }
+
+        pub fn longName(m: Month) []const u8 {
+            return names[@intFromEnum(m) - 1];
+        }
+    };
+
+    /// Get current date in the system's local time.
+    pub fn now(allocator: std.mem.Allocator) LocalTimeError!Date {
+        const utc_timestamp = timestamp();
+
+        if (builtin.os.tag == .windows) {
+            var tzi: std.os.windows.TIME_ZONE_INFORMATION = undefined;
+            const rc = std.os.windows.ntdll.NtQuerySystemInformation(
+                .SystemCurrentTimeZoneInformation,
+                &tzi,
+                @sizeOf(std.os.windows.TIME_ZONE_INFORMATION),
+                null,
+            );
+            if (rc != .SUCCESS) return error.LocalTimeUnavailable;
+            return fromTimestamp(utc_timestamp, .{
+                .name_data = .{
+                    @intCast(tzi.StandardName[0]),
+                    @intCast(tzi.StandardName[1]),
+                    @intCast(tzi.StandardName[2]),
+                    @intCast(tzi.StandardName[3]),
+                    @intCast(tzi.StandardName[4]),
+                    @intCast(tzi.StandardName[5]),
+                },
+                .flags = 0,
+                .offset = tzi.Bias,
+            });
+        }
+
+        var sf = std.heap.stackFallback(4096, allocator);
+        var tz = try localtime(sf.allocator());
+        defer tz.deinit(sf.allocator());
+
+        return fromTimestamp(utc_timestamp, tz.project(utc_timestamp));
+    }
+
+    /// Convert timestamp in milliseconds to a Date.
+    pub fn fromTimestamp(utc_timestamp: i64, tt: TimeZone.TimeType) Date {
```

Comment:
Why in milliseconds?
If we're going to pick a fraction of a second, shouldn't it be nanoseconds?

Reply:
For one, "UTC" is misleading. Unix time is not UTC, since it discards leap seconds. 

Second, why not use a variable precision, as @karlseguin does in [zul](https://github.com/karlseguin/zul/blob/master/src/datetime.zig#L227) ?